app/Providers/SongStorageServiceProvider.php

```
@@ -18,10 +24,26 @@ public function register(): void
                 'sftp' => SftpStorage::class,
+                'webdav' => WebDAVStorage::class,
                 default => LocalStorage::class,
             };
 
             return $this->app->make($concrete);
         });
     }
+
+    public function boot(): void
+    {
+        Storage::extend('webdav', static function ($app, array $config): FilesystemAdapter {
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$app'. (undefined)

(UnusedFormalParameter)